`make doc` doesn't work immediately after nightly upgrade.

If the nightly version changes and you run `make doc` rather than `make` things seem to not work.

```
$ make doc
error: toolchain 'nightly-2017-06-19' is not installed
info: caused by: not a directory: '/Users/bradjc/.rustup/toolchains/nightly-2017-06-19-x86_64-apple-darwin'
info: syncing channel updates for 'nightly-2017-06-19-x86_64-apple-darwin'
info: downloading component 'rustc'
info: downloading component 'rust-std'
info: downloading component 'cargo'
info: downloading component 'rust-docs'
info: installing component 'rustc'
info: installing component 'rust-std'
info: installing component 'cargo'
info: installing component 'rust-docs'
error: `rust-src` component not found. Run `rustup component add rust-src`.
note: run with `RUST_BACKTRACE=1` for a backtrace
make: *** [doc] Error 1

$ rustup component add rust-src
info: downloading component 'rust-src'
 27.4 MiB /  27.4 MiB (100 %)   6.1 MiB/s ETA:   0 s
info: installing component 'rust-src'

$ make doc
error: `rust-src` component not found. Run `rustup component add rust-src`.
note: run with `RUST_BACKTRACE=1` for a backtrace
make: *** [doc] Error 1
```

--- a/boards/Makefile.common
+++ b/boards/Makefile.common
@@ -41,6 +41,7 @@ ifneq ($(shell RUSTUP_TOOLCHAIN=$(RUSTUP_TOOLCHAIN) rustc --version), $(RUSTC_VE
     $(warning You may experience unexpected compilation warnings or errors)
     $(warning To fix, install `rustup` from https://rustup.rs/, then run: `rustup install $(RUSTUP_TOOLCHAIN)`)
   endif
+  $(shell RUSTUP_TOOLCHAIN=$(RUSTUP_TOOLCHAIN) rustup component add rust-src)
 endif
 
 # Check that xargo is installed
@@ -88,7 +89,6 @@ target/$(TARGET)/release/$(PLATFORM).elf: target/$(TARGET)/release/$(PLATFORM)
 
 .PHONY: target/$(TARGET)/release/$(PLATFORM)
 target/$(TARGET)/release/$(PLATFORM):
-	$(Q)rustup component add rust-src
 	$(Q)RUSTFLAGS=$(RUSTFLAGS_FOR_XARGO_LINKING) $(XARGO) build --target=$(TARGET) $(VERBOSE) --release
 	$(Q)$(SIZE) $@
 
@@ -117,7 +117,6 @@ target/$(TARGET)/debug/$(PLATFORM).bin: target/$(TARGET)/debug/$(PLATFORM).elf
 # binary. This makes checking for Rust errors much faster.
 .PHONY: check
 check:
-	$(Q)rustup component add rust-src
 	$(Q)RUSTFLAGS=$(RUSTFLAGS_FOR_XARGO_LINKING) $(XARGO) check --target=$(TARGET) $(VERBOSE) --release
 
 .PHONY: clean

